src/auth/oauth.ts

```diff
@@ -554,4 +582,60 @@ export class OAuth extends BaseAuthAPI {
 
     return VoidApiResponse.fromResponse(response);
   }
+
+  /**
+   * Exchanges a subject token (e.g. a refresh token) for a federated connection access token.
+   *
+   * The request body includes:
+   * - client_id (and client_secret via addClientAuthentication)
+   * - grant_type set to FCAT_GRANT_TYPE
+   * - subject_token and fixed subject_token_type value for refresh tokens
+   * - requested_token_type indicating that a federated connection access token is desired
+   * - connection name and an optional login_hint if provided
+   *
+   * @param options - The federated connection options.
```

Comment:
I think we try and avoid `federated` in the public API. Perhaps `The options to retrieve a token for a connection` or so.